src/diagnostics/MOM_harmonic_analysis.F90

```diff
@@ -0,0 +1,411 @@
+!> Inline harmonic analysis (conventional)
+module MOM_harmonic_analysis
+
+use MOM_time_manager,  only : time_type, real_to_time, time_type_to_real, get_date, increment_date, &
+                              operator(+), operator(-), operator(<), operator(>), operator(>=)
+use MOM_grid,          only : ocean_grid_type
+use MOM_unit_scaling,  only : unit_scale_type
+use MOM_file_parser,   only : param_file_type, get_param
+use MOM_io,            only : file_exists, open_ASCII_file, READONLY_FILE, close_file, &
+                              MOM_infra_file, vardesc, MOM_field, &
+                              var_desc, create_MOM_file, SINGLE_FILE, MOM_write_field
+use MOM_error_handler, only : MOM_mesg, MOM_error, NOTE
+
+implicit none ; private
+
+public HA_init, HA_register, HA_accum
+
+#include <MOM_memory.h>
+
+integer, parameter :: MAX_CONSTITUENTS = 10  !< The maximum number of tidal constituents
+
+!> The private control structure for storing the HA info of a particular field
+type, private :: HA_type
+  character(len=16) :: key = "none"          !< Name of the field of which harmonic analysis is to be performed
+  real :: old_time = -1.0                    !< The time of the previous accumulating step
```

Comment:
Please modify the comments describing the units of the real variables on lines 25 and 27-29.  If these are used in various ways, this description might be `[various]`, but if you could give a specific example (e.g., for SSH) those units would also help in making sense of the intended purpose of these variables.